kitchen-tests/cookbooks/end_to_end/recipes/_habitat_service.rb

```diff
@@ -0,0 +1,311 @@
+habitat_sup "default" do
+  hab_channel "stable"
+  license "accept"
+  gateway_auth_token "secret"
+end
+
+ruby_block "wait-for-sup-default-startup" do
```

Comment:
shouldn't this be part of the main resource so it blocks on starting?

Reply:
Notes added. The main service block does handle waiting to an extent. Habitat itself is the hang up here. We cannot account for the way a service is configured and installed so load times get very sporadic. This particular recipe loads, unloads, and restarts these services multiple times. In a pipeline those times were too unpredictable.